src/Stratis.Bitcoin/BlockPulling/BlockPullerBehavior.cs

```diff
@@ -197,14 +197,18 @@ private Task OnMessageReceivedAsync(INetworkPeer peer, IncomingMessage message)
         {
             if (message.Message.Payload is BlockPayload block)
             {
+                block.Obj.Header.PrecomputeHash(true, true);
```

Comment:
Just an observation - there is potential for a DoS vector here, as the entire block & all transactions are getting hashed before any rules get applied to it.

Perhaps the risk is minor though, as ostensibly we will only request blocks for which we already have a valid header.

@noescape00 what do you think?

Reply:
No the PrecomputeHash method has an option to tell it to lazy load, so only when the hash is calculated we put it in cache.

Reply:
Ah ok, I made an (incorrect) assumption about what `PrecomputeHash` does based on the name.

Reply:
>as the entire block & all transactions are getting hashed before any rules get applied to it.

only stuff from header gets hashed. Txes affect merkle tree root which is inside of a header.
so should be ok to do it.